app/src/main/java/com/amaze/filemanager/ui/dialogs/SmbConnectDialog.java

```diff
@@ -350,6 +354,9 @@ private SmbFile createSMBPath(String[] auth, boolean anonymous, boolean disableI
             .append(URLEncoder.encode(auth[2], "UTF-8"))
             .append("@");
       sb.append(yourPeerIP).append("/");
+      if (share != null || share.length() > 0) {
```

Comment:
You may simply to use [TextUtils.isEmpty](https://developer.android.com/reference/android/text/TextUtils#isEmpty(java.lang.CharSequence)) for string empty/null check.

Reply:
Perfect, thanks for the tip (I am not an android dev by any stretch).... I will take care of those formatting issues as well tonight and hopefully we will have something acceptable by tomorrow :)

Thank you for the help and guidance!

Reply:
By using `!TextUtils.isempty(share)` I believe this should fix the extra slash issue too (because of the `||` instead of `&&` in the original code, it was adding the slash even on a null/empty value). Though I can't double check this because I don't have any non-hidden shares :(